src/h5web/visualizations/heatmap/utils.ts

```diff
@@ -17,3 +17,23 @@ export function generateCSSLinearGradient(
 
   return `linear-gradient(to ${direction},${gradientColors})`;
 }
+
+export function getPixelEdges(
+  pixelCoordinates: number[] | undefined,
+  nPixels: number
+): number[] {
+  if (!pixelCoordinates) {
+    return range(nPixels + 1);
+  }
+
+  if (pixelCoordinates.length === nPixels) {
+    // Add the last edge assuming the pixelCoordinates are regularly spaced
+    const deltaCoord = pixelCoordinates[1] - pixelCoordinates[0];
+    return [
+      ...pixelCoordinates,
+      pixelCoordinates[pixelCoordinates.length - 1] + deltaCoord,
+    ];
+  }
+
+  return pixelCoordinates;
```

Comment:
I could move the `throw` here but I prefer to return `pixelCoordinates` and let the caller deal with it.

Reply:
Hmm I feel like putting the throw here would be clearer, since it's the unsupported case that follows our three supported cases: `!pixelCoordinates`, `pixelCoordinates.length ===  numPixels + 1` and `pixelCoordinates.length ===  numPixels`.

I would actually add an if/return between the two blocks you already have, to identify clearly the second case I list above, `pixelCoordinates.length ===  numPixels + 1`. The final line can then become the `throw`, which should explain that the input pixel coordinates have an unsupported length.